src/resolver.rs

```diff
@@ -472,7 +581,7 @@ impl CriteriaMapper {
                     cur_idx == other_idx || !self.implied_criteria[other_idx].has_criteria(cur_idx)
                 })
             })
-            .map(|idx| &*self.list[idx].0)
+            .map(|idx| &*self.list[idx].1)
```

Comment:
Note to self, in principle this should get some kind of disambiguating prefix like `format("{.0}::{.1}")` in the case of foreign criteria. I'm actually inclined to implement this as inserting that preformatted string into `self.list[..].1` itself, and making other code `split_once`. That or we add two strings to the list, one for the "local name" and one for the "global name". Now that I'm saying it that sounds a lot simpler lol.

Reply:
Oh other note to self, this code is completely busted. The `!implied.has_criteria` check is fundamentally assuming non-circularity. We should tiebreak circularity by bypassing this guard when cur_idx is `Local` and other_idx` is `Foreign`.

Reply:
Final note to self, different code probably wants different behaviour on showing/hiding foreign criteria. But needs experimentation.